protocol.md

```diff
@@ -870,6 +902,12 @@ export interface TextDocumentClientCapabilities {
 ```typescript
 interface ClientCapabilities {
 	/**
+	 * The client supports receiving the result solely through $/partialResult notifications for requests from the client to the server.
+	 * If true, the server should send all data through $/partialResult and send null as the result in the final response.
```

Comment:
@nicksnyder 

> If a client sends a request A and a request B, and the server finished processing request B first, there is nothing about JSON RPC that prevents the server from sending the response for request B before the response for request A.

I never meant the server is not allowed to send the response for B before the response A. I'm just saying if the request to A was received before the request to B, the server should trigger the method call to A before the method call to B. The methods can be executed in parallel and return at arbitrary times.

Imo making sure no packages get lost and they arrive in the correct order is task of the transport level protocol (TCP), not the application level protocol (LSP).